telegram/src/main/kotlin/com/github/kotlintelegrambot/dispatcher/handlers/NewChatMembersHandler.kt

```diff
@@ -1,15 +1,29 @@
 package com.github.kotlintelegrambot.dispatcher.handlers
 
+import com.github.kotlintelegrambot.Bot
+import com.github.kotlintelegrambot.HandleNewChatMembers
 import com.github.kotlintelegrambot.HandleUpdate
 import com.github.kotlintelegrambot.entities.Update
 
 class NewChatMembersHandler(
-    handlerCallback: HandleUpdate
-) : Handler(handlerCallback) {
+    handleNewChatMembers: HandleNewChatMembers
+) : Handler(NewChatMembersHandlerProxy(handleNewChatMembers)) {
     override val groupIdentifier: String
-        get() = "System"
+        get() = "newChatMembers"
 
     override fun checkUpdate(update: Update): Boolean {
         return update.message?.newChatMembers != null
     }
 }
+
+private class NewChatMembersHandlerProxy(
+    val handleNewChatMembers: HandleNewChatMembers
+) : HandleUpdate {
+    override fun invoke(bot: Bot, update: Update) {
+        val message = update.message
+        val newChatMembers = message?.newChatMembers
+        checkNotNull(message)
+        checkNotNull(newChatMembers)
+        handleNewChatMembers.invoke(bot, update.message, newChatMembers)
```

Comment:
Not really important, but you could use the local `message` val directly as argument for `invoke`. Also, I think it's enough to call `checkNotNull` just with `newChatMembers`; the compiler will infer that message is not null either.

Reply:
sure, i hurried